goodkey/good_key.go

```diff
@@ -40,15 +40,24 @@ type KeyPolicy struct {
 	AllowRSA           bool // Whether RSA keys should be allowed.
 	AllowECDSANISTP256 bool // Whether ECDSA NISTP256 keys should be allowed.
 	AllowECDSANISTP384 bool // Whether ECDSA NISTP384 keys should be allowed.
+	weakRSAList        *weakKeys
 }
 
 // NewKeyPolicy returns a KeyPolicy that allows RSA, ECDSA256 and ECDSA384.
-func NewKeyPolicy() KeyPolicy {
-	return KeyPolicy{
+func NewKeyPolicy(weakKeyDir string) (KeyPolicy, error) {
```

Comment:
*weakKeyFile. Also, please add to the function comment describing the param, and the behavior when it's empty.